client/src/main/scala/org/http4s/client/Client.scala

```diff
@@ -26,8 +26,7 @@ final case class DisposableResponse[F[_]](response: Response[F], dispose: F[Unit
     * HTTP connection when the task finishes.
     */
   def apply[A](f: Response[F] => F[A])(implicit F: MonadError[F, Throwable]): F[A] = {
-    val task = try f(response)
-    catch { case e: Throwable => F.raiseError[A](e) }
+    val task = f(response)
```

Comment:
This is specifically the change I want to discuss.

I'm unsure whether catching this is.. good. It's a sort of cushion but I do not think this should be necessary if the result is parametrized to begin with.

Reply:
An exception here will leak a connection, which is a severe response to a newbie mistake.  A try that is uncaught is negligible, to possibly zero, performance overhead.  (I've been told both and not measured myself.)  I think this might be a good case to lecture in the logs but try to keep the system operational.

Reply:
Fair.. I just wanted to discuss it and leave a note there in the case we do not remove it.